src/validator.cpp

```diff
@@ -273,6 +273,38 @@ struct Validator::ValidatorImpl
      */
     bool isSupportedMathMLElement(const XmlNodePtr &node);
 
+    /**
+    * @brief Validate that equivalent variable pairs in the @p model
+    * have equivalent units.
+    * Any issues will be logged in the @c Validator.
+    *
+    * Any difference in base units is reported as an issue in the @c Validator, but the multiplier difference does not trigger a validator issue.
+    * Where the base units are equivalent, the multiplier may be interpreted as units_of_v1 = (10^multiplier)*units_of_v2
+    *
+    * @param model The model containing the variables
+    * @param v1 The variable which may contain units.
+    * @param v2 The equivalent variable which may contain units.
+    * @param hints String containing issue messages to be passed back to the calling function for logging.
+    * @param multiplier Double returning the effective multiplier mismatch between the units.
+    */
+    // KRM bool unitsAreEquivalent(const ModelPtr &model, const VariablePtr &v1, const VariablePtr &v2, std::string &hints, double &multiplier);
+
+    /**
+    * @brief Utility function used by unitsAreEquivalent to compare base units of two variables.
+    *
+    * @param model The model containing the variables.
+    * @param unitMap A list of the exponents of base variables.
+    * @param uName String name of the current variable being investigated.
+    * @param standardList Nested map of the conversion between built-in units and the base units they contain
+    * @param uExp Exponent of the current unit in its parent.
+    * @param direction Specify whether we want to increment (1) or decrement (-1).
+    */
+    void updateBaseUnitCount(const ModelPtr &model,
+                             std::map<std::string, double> &unitMap,
+                             double &multiplier,
+                             const std::string &uName,
+                             double uExp, double logMult, int direction);
```

Comment:
Is that needed? What about lines 54-68?

Reply:
I've removed this one in favour of the earlier one.  This will likely change soon as I think we should rather use the functions in the `Units` class, but that's a story for another PR!